traitlets/traitlets.py

```diff
@@ -649,14 +697,23 @@ def observe(*names):
     """
     return ObserveHandler(names)
 
+def validate(name):
+    """ A decorator which can be used to cross validate a member on a class.
```

Comment:
Let's not use the word "cross validate".  How about "A decorator for validating the state of an HasTraits object when a Trait is set." or something like that.